packages/cdktf/lib/terraform-asset.ts

```diff
@@ -0,0 +1,102 @@
+import { Construct, ISynthesisSession, Node } from "constructs";
+import * as fs from "fs";
+import * as path from "path";
+import { Manifest } from "./manifest";
+import { copySync, hashPath } from "./private/fs";
+import { Resource } from "./resource";
+
+export interface TerraformAssetConfig {
+  readonly path: string;
+  readonly type?: AssetType;
+  readonly assetHash?: string;
+}
+
+export enum AssetType {
+  FILE,
+  DIRECTORY,
+}
+
+const ASSETS_DIRECTORY = "assets";
+
+export class TerraformAsset extends Resource {
+  private sourcePath: string;
+  public assetHash: string;
```

Comment:
I can see how having this could be useful, but there doesn't appear to be any use of this currently.

Reply:
it's intended to be used by resources that rely on assets, e.g. a lambda resource could choose not to recreate because the hash is the same